app/src/cose.h

```diff
@@ -396,17 +804,27 @@ namespace scitt::cose
 
     QCBOREncode_OpenArray(encoder);
 
-    QCBOREncode_AddBytes(encoder, protected_header);
+    QCBOREncode_AddBytes(encoder, cbor::from_bytes(protected_header));
 
     // unprotected header
     QCBOREncode_OpenMap(encoder);
     QCBOREncode_OpenArrayInMapN(encoder, COSE_HEADER_PARAM_SCITT_RECEIPTS);
     QCBOREncode_AddEncoded(encoder, cbor::from_bytes(receipt));
     QCBOREncode_CloseArray(encoder);
+    if (x5chain.has_value())
```

Comment:
If `x5chain` only has one element it should be serialized as a single bstr, not an array.

https://datatracker.ietf.org/doc/html/draft-ietf-cose-x509-09 Section 2
> If a single certificate is conveyed, it is placed in a CBOR byte string.
         
And in Table 1: `COSE_X509 = bstr / [ 2*certs: bstr ]`, where `2*` means it needs at least two elements.

Reply:
Done in https://github.com/microsoft/scitt-ccf-ledger/pull/90